### PR TITLE
Fix/compiler warnings

### PR DESCRIPTION
**Pull request by cwitting**
@helenol @mfehr So voxblox compiles without warnings using -Wall (only some weird catkin warning). However all these cast are probably only symptom handling and not the right way to do it. 
But the other fixes should be legit.

### CODE REVIEW

**Comment by helenol** [comment]
add to whitelist

**Comment by ZacharyTaylor** [comment]
Thanks for going through and sorting these warnings. One issue though is that we usually try to stick to the google style guide and the use of c style castings goes against this https://google.github.io/styleguide/cppguide.html#Casting could you swap them to static_cast ?

File: voxblox/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ find_package(catkin_simple REQUIRED)
 catkin_simple()
 
 set(CMAKE_MACOSX_RPATH 0)
-add_definitions(-std=c++11 -Wno-sign-compare -Wno-unused-value)
+add_definitions(-std=c++11 -Wall)
```
Comment (mfehr) [inline voxblox/CMakeLists.txt]: Awesome :+1:

File: voxblox/src/core/esdf_map.cc [hunk 2]
```diff
@@ -17,7 +17,7 @@ bool EsdfMap::getDistanceAtPosition(const Eigen::Vector3d& position,
 bool EsdfMap::getDistanceAndGradientAtPosition(
     const Eigen::Vector3d& position, double* distance,
     Eigen::Vector3d* gradient) const {
-  FloatingPoint distance_fp;
+  FloatingPoint distance_fp = 0;
```
Comment (mfehr) [inline voxblox/src/core/esdf_map.cc]: 0.0

**Pull request merged**